flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/runtime/stream/sql/join/JoinReorderITCase.java

```diff
@@ -0,0 +1,465 @@
+import org.apache.flink.table.api.Table;
+import org.apache.flink.table.api.bridge.java.StreamTableEnvironment;
+import org.apache.flink.table.api.config.ExecutionConfigOptions;
+import org.apache.flink.table.api.config.OptimizerConfigOptions;
+import org.apache.flink.table.catalog.Catalog;
+import org.apache.flink.table.catalog.ObjectPath;
+import org.apache.flink.table.catalog.stats.CatalogColumnStatistics;
+import org.apache.flink.table.catalog.stats.CatalogColumnStatisticsDataBase;
+import org.apache.flink.table.catalog.stats.CatalogColumnStatisticsDataLong;
+import org.apache.flink.table.catalog.stats.CatalogTableStatistics;
+import org.apache.flink.table.planner.factories.TestValuesTableFactory;
+import org.apache.flink.table.planner.runtime.batch.sql.join.JoinITCaseHelper;
+import org.apache.flink.table.planner.runtime.batch.sql.join.JoinType;
+import org.apache.flink.table.planner.runtime.utils.TestData;
+import org.apache.flink.table.planner.runtime.utils.TestingRetractSink;
+import org.apache.flink.table.planner.utils.JavaScalaConversionUtil;
+import org.apache.flink.test.util.TestBaseUtils;
+import org.apache.flink.types.Row;
+import org.apache.flink.util.CollectionUtil;
+import org.apache.flink.util.TestLogger;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import scala.Enumeration;
+import scala.Tuple2;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+/**
+ * ITCase for JoinReorder in stream mode. This class will test {@link
+ * org.apache.flink.table.planner.plan.rules.logical.FlinkBushyJoinReorderRule} and {@link
+ * org.apache.calcite.rel.rules.LoptOptimizeJoinRule} together by changing the factor
+ * isBushyJoinReorder.
+ */
+@RunWith(Parameterized.class)
+public class JoinReorderITCase extends TestLogger {
+
+    private StreamExecutionEnvironment env;
+    private StreamTableEnvironment tEnv;
+    private Catalog catalog;
+
+    @Parameterized.Parameter(value = 0)
+    public Enumeration.Value expectedJoinType;
+
+    @Parameterized.Parameter(value = 1)
+    public boolean isBushyJoinReorder;
+
+    @Parameterized.Parameters(name = "expectedJoinType={0}, isBushyJoinReorder={1}")
+    public static Collection<Object[]> data() {
+        return Arrays.asList(
+                new Object[][] {
+                    {JoinType.BroadcastHashJoin(), false},
+                    {JoinType.BroadcastHashJoin(), true},
+                    {JoinType.HashJoin(), false},
+                    {JoinType.HashJoin(), true},
+                    {JoinType.SortMergeJoin(), false},
+                    {JoinType.SortMergeJoin(), true},
+                    {JoinType.NestedLoopJoin(), false},
+                    {JoinType.NestedLoopJoin(), true}
```

Comment:
there are no broadcast join, hash join, sort merge join, and nested loop join for streaming

Reply:
> there are no broadcast join, hash join, sort merge join, and nested loop join for streaming

Done!